controllers/image_generator.go

```diff
@@ -59,6 +59,9 @@ const (
 	LibvirtProvider               = "libvirt"
 	peerpodsImageJobsPathLocation = "/config/peerpods/podvm"
 	azureImageGalleryPrefix       = "PodVMGallery"
+	preBuiltImageType             = "pre-built"
+	operatorBuiltImageType        = "operator-built"
+	libvirtPeerPodsCMName         = "libvirt-podvm-image-cm"
```

Comment:
generic comment on code organisation: I would prefer the upload aspects are contained in image_uploader.go file

Reply:
Also I don't understand why you need uploader? 
Is uploader coupled with using pre-built image or it's something additional to make it easier to upload the built image to libvirt volume and applicable for both pre-built image and the one created by the operator?

Reply:
You can refactor the following instructions in create_libvirt_image function https://github.com/openshift/sandboxed-containers-operator/blob/devel/config/peerpods/podvm/libvirt-podvm-image-handler.sh#L43-L60
and move it out to a different function (eg. create_libvirt_image_from_scratch). 
Add another function to handle pre-built image (eg create_libvirt_image_from_prebuilt_artifact) 
IOW, create_libvirt_image will call either create_libvirt_image_from_scratch or create_libvirt_image_from_prebuilt_artifact depending on the value of PODVM_IMAGE_URI
Then for uploading the image you'll still continue to use existing logic in https://github.com/openshift/sandboxed-containers-operator/blob/devel/config/peerpods/podvm/libvirt-podvm-image-handler.sh#L62-L69

Add all functions to handle the PODVM_IMAGE_URI (parsing etc) in lib.sh or create a separate file to keep the helper functions.

This is how I think it should be structured. Of course this is a suggestion and feel free to improvise.
However, I don't see any need for a separate uploader job for now.

Reply:
> Also I don't understand why you need uploader?

Idea was to keep separate jobs for creation/upload and standalone uploading, so that it is easy to maintain individually and as you pointed it makes the job easier to upload the built image to libvirt volume whereasit was not planned for both the cases, just for the upload scenario.

I will incorporate the suggestions you have given.
Thanks Pradipta.